presto-main/src/main/java/com/facebook/presto/operator/TaskStats.java

```diff
@@ -159,13 +159,13 @@ public TaskStats(
 
             @JsonProperty("cumulativeUserMemory") double cumulativeUserMemory,
             @JsonProperty("cumulativeTotalMemory") double cumulativeTotalMemory,
-            @JsonProperty("userMemoryReservation") long userMemoryReservationInBytes,
+            @JsonProperty("userMemoryReservationInBytes") long userMemoryReservationInBytes,
```

Comment:
I don't think these names have to match up with the variable name.
whatever is in `@JsonProperty` takes precedence.
I'm sure there are plenty of examples of this in the code base.
One I could find with a quick grep.
https://github.com/prestodb/presto/blob/9e46a70a4be0e9b2a6ffc55a7f341947299f5e27/presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/JdbcSplit.java#L52

Reply:
@ajaygeorge I thought the same and the code passes all tests from maven and even works in production. But the related tests cases filed internally with buck test. Then I tried to review the code to see what's going on. I try to bring the airlift and jackson libraries to the same level but to no avail. Then I start reviewing the code to see what happen and tried to apply the change you see then the test cases passed. So to this point i have no idea why the same code works elsewhere. Also the other issue in the PR about JsonCreator annotation, the existing code is obviously not compliant with the jackson doc (see the PR descriptions for details) but the code works. 
Since the change is not harmful, let's land it first. I will see how to take case of similar code later.